src/core/structure/sheet.ts

```diff
@@ -81,6 +81,87 @@ export default class Sheet {
     return cell ? cell.value : null;
   }
 
+  moveColumn(from: number, to: number): boolean {
+    if (from === to) return false;
+
+    const colKey = this.columnPositions.get(from);
+
+    if (colKey !== undefined) {
+      const col = this.columns.get(colKey);
+      if (col === undefined)
+        throw new Error("Column not found, inconsistent state");
+      col.position = to;
+    }
+
+    //we have to update all the positions to keep it consistent
+    this.columnPositions.delete(from);
+    if (to > from) {
+      this.shiftColumns(to, ShiftDirection.backward);
+    } else {
+      this.shiftColumns(to, ShiftDirection.forward);
+    }
+    if (colKey === undefined) {
+      this.columnPositions.delete(to);
```

Comment:
In what case is this delete operation needed?

Reply:
It is needed if we are trying to move an empty column to a different location

Reply:
When shifting to the left I suppose? When shifting to the right, the position `to` should always be left empty.
I noticed now that you're shifting left when `from < to`, but shouldn't that also be a shift to the right?
For example, in the case below, I would expect moving A to between B and C (position 2) to push C and D to the right, not B to the left. This behavior is up to us to define of course, but I think this would be more consistent.
![image](https://github.com/lightsheet-team/lightsheet/assets/38987331/071105b6-5867-4e2c-b52b-458cca634380)

Reply:
> I would expect moving A to between B and C (position 2) to push C and D to the right, not B to the left

Well, I did it like excel and sheet, which is shifting B to the left. 
I also feel like it better that way because I imagine that the purpose of moving columns is ordering, and if we shift C and D to the right we are like doing an implicit insert and then there will be a gap in A, which means I would need to do another move operation to get them in the place I want them.

Reply:
Makes sense :+1: I couldn't figure out how to do the same action in Excel :P